src/DataTransferObjectCollection.php

```diff
@@ -7,6 +7,7 @@
 use Iterator;
 use Countable;
 use ArrayAccess;
+use Spatie\DataTransferObject\Contracts\DtoContract;
```

Comment:
This extra contract seems like an overcomplication. Can you explain the reasoning behind it?

Reply:
Until php has better type variance support this seemed like a good idea to sprcify the return type of a dto. I just added it everywhere for consistency